Automattic/jetpack
# Protect: Add onboarding dialog

## Description:
Adds onboarding dialog

## Proposed changes:

* Adds a custom `useOnboardingPopover` hook with `OnboardingRenderedContext`, used within a new `OnboardingPopover` component, implemented on the `ScanPage`
* Uses the `ActionPopover` js-packages component to add conditional onboarding flows to the Protect UI

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/331

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Start up your Jurassic Tube and test the following flows (mobile and desktop)
* Ensure no regressions are introduced

**Free with or without threats**
- Your scan results
- Daily automated scanning
- Dismiss or Finish onboarding and verify that the popover no longer load
- Upgrade and proceed
![Screen Capture on 2023-12-18 at 15-16-20](https://github.com/Automattic/jetpack/assets/43220201/0c49b2fb-6a90-4622-a438-da1d1880c3e4)

**Paid with no threats**
- Your scan results
- Daily automated and manual scanning
![Screen Capture on 2023-12-18 at 16-58-42](https://github.com/Automattic/jetpack/assets/43220201/3047a91b-f2c9-41b1-b1ce-ba2b66a95048)

**Paid with non-fixable threats**
- Your scan results
- Understand severity
- Daily automated and manual scanning
![Screen Capture on 2023-12-18 at 15-30-35](https://github.com/Automattic/jetpack/assets/43220201/ba65017f-9ccb-45ec-af57-7bf22f66aedd)

**Paid with fixable threats**
- Your scan results
- Auto-fix all threats
- Understand severity
- Daily automated and manual scanning
- Dismiss or Finish onboarding and verify that the popovers no longer load
![Screen Capture on 2023-12-18 at 16-56-20](https://github.com/Automattic/jetpack/assets/43220201/bb872a09-5201-4b42-a49e-18899b2c2372)
